nexus/src/app/sagas/disk_create.rs

```diff
@@ -51,8 +51,16 @@ lazy_static! {
         sdc_create_disk_record,
         sdc_create_disk_record_undo
     );
-    static ref REGIONS_ALLOC: NexusAction =
-        new_action_noop_undo("disk-create.regions-alloc", sdc_alloc_regions,);
+    static ref REGIONS_ALLOC: NexusAction = ActionFunc::new_action(
+        "disk-create.allocate-regions",
+        sdc_alloc_regions,
+        sdc_alloc_regions_undo,
```

Comment:
While I was here...

I noticed region allocation didn't have an undo operation, so I added one.

Reply:
I don't understand where the act of "undoing region allocation" could be *other* than the undo action of the node that allocated those regions.

If we start performing the subsequent operation (which, in the current DAG, is `regions_ensure`) and we fail for *any* reason, the next executed node would be "undo the `REGIONS_ALLOC` node". If the undo operation isn't here, where could it be?

Reply:
All that being said: If `sdc_create_volume_record_undo` is *also* de-allocating these regions, I think the PR is incorrect as-is. However, I think it should only be responsible for removing the volume record, not de-allocating regions, for the reasons listed above (we could fail before that point).

Reply:
We chatted about this, please disregard my previous comment. I was concerned about a scenario that isn't possible, and even if it was this isn't the way to fix it.